tests/test_execution.py

```diff
@@ -317,36 +272,20 @@ def test_aggregate_composite_traintuples(factory, session_1, session_2):
 
     This test refers to the model composition use case.
     """
-    aggregate_worker = session_1.node_id
-    sessions = [session_1, session_2]
+    factory, network = data_network
+    sessions = network.sessions
+
+    aggregate_worker = sessions[0].node_id
     number_of_rounds = 2
 
-    # register objectives, datasets, and data samples
-    datasets = []
-    for s in sessions:
-        # register one dataset per node
-        spec = factory.create_dataset()
-        dataset = s.add_dataset(spec)
-        datasets.append(dataset)
-
-        # register one data sample per dataset per round of aggregation
-        for _ in range(number_of_rounds):
-            spec = factory.create_data_sample(test_only=False, datasets=[dataset])
-            s.add_data_sample(spec)
     # reload datasets (to ensure they are properly linked with the created data samples)
+    datasets = sessions[0].state.datasets + sessions[1].state.datasets
     datasets = [
         sessions[i].get_dataset(d.key)
         for i, d in enumerate(list(datasets))
     ]
```

Comment:
This should be done in the `data_network` fixture.

Reply:
👍